noobaa/noobaa-operator
# Update Package Cloud Credential Operator (CCO)

### Explain the changes
1. Update package cloud-credential-operator, by running: `go get -u github.com/openshift/cloud-credential-operator`, the output is:
```bash
go: downloading github.com/openshift/cloud-credential-operator v0.0.0-20231004191224-abdf0627a0cf
go: upgraded github.com/asaskevich/govalidator v0.0.0-20200907205600-7a23bdc65eef => v0.0.0-20210307081110-f21760c49a8d
go: upgraded github.com/openshift/cloud-credential-operator v0.0.0-20210716145643-0ee1808fb8ed => v0.0.0-20231004191224-abdf0627a0cf
go: upgraded k8s.io/kube-aggregator v0.19.0 => v0.27.1
```
3. Changes in code related to the update of the package of the cloud credential operator.
change `NewCodec()` to `Codec` since it was removed and we should use global codec (for more information see PR openshift/cloud-credential-operator#576.

### Issues: Fixed #xxx / Gap #xxx
1. none

### Testing Instructions:
1. Run `make all`.
Note: before the code change we had 2 errors (same error message):
```bash
undefined: cloudcredsv1.NewCodec
```
2. Use the cluster bot using `launch 4.14 aws`, then noobaa install should be Ready and we should have default backingstore type will be aws-s3, inside the logs of the operator we should have the printing "_Secret noobaa-aws-cloud-creds-secret was created successfully by cloud-credentials operator_".
